server/network/aoprotocol.py

```diff
@@ -614,8 +621,12 @@ def net_cmd_ms(self, args):
                                             )
 
         self.client.area.set_next_msg_delay(len(msg))
-        database.log_ic(self.client, self.client.area, showname, msg)
-
+        if(bool(self.server.config['log_chat'])):
```

Comment:
This check will fail with a KeyError if `log_chat` is not defined in the config. You need to provide a sane default value in the config load function.